jarviscli/plugins/voice_control.py

```diff
@@ -15,27 +17,50 @@
         'voice_control_requirements (install portaudio + re-run setup.sh)']
 
 
+def has_internet():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    
+    try:
+        socket.setdefaulttimeout(3)
+        sock.connect(('8.8.8.8', 8000))
+
+        return True
+    
+    except socket.timeout:
+        return False
+
+
 @require(native=requirements)
 @plugin("hear")
 def hear(jarvis, s):
     r = sr.Recognizer()  # intializing the speech_recognition
     listen = False
+
     _jarvis = jarvis._jarvis  # calling jarvis object.
     _jarvis.speech.text_to_speech("Say listen to start voice mode")
+
+    connected = has_internet()
+
     while listen is False:
         try:
             with sr.Microphone() as source:
                 os.system('reset')  # for clearing the terminal.
                 print("Say listen to start listening")
                 r.adjust_for_ambient_noise(source)  # Eleminating the noise.
                 audio = r.listen(source)  # Storing audio.
-                pinger = r.recognize_google(audio)  # Converting speech to text
+
+                if connected:
+                    pinger = r.recognize_google(audio)  # Converting speech to text using google recognition.
+                
+                else:
+                    pinger = r.recognize_sphinx(audio)  # Converting speech to text using Sphinx CMU in case user is not connected to internet
```

Comment:
Probably needs a check if sphinx (pip package: pocketsphinx) is installed. And we should add pocketsphinx as an optional requirement to our installer.

Reply:
i had a bit of a pain installing pocketsphinx with pip. im not exactly sure why, but installing directly from a wheel worked for me. 

thats definitely something I'll look into though.